lib/wanda/execution/server.ex

```diff
@@ -125,15 +124,21 @@ defmodule Wanda.Execution.Server do
     if Gathering.all_agents_sent_facts?(agents_gathered, targets) do
       result = Evaluation.execute(execution_id, group_id, checks, gathered_facts)
 
-      execution_completed = Messaging.Mapper.to_execution_completed(result)
-      :ok = Messaging.publish("results", execution_completed)
+      store_and_publish_execution_result(result)
 
       {:stop, :normal, state}
     else
       {:noreply, state}
     end
   end
 
+  defp store_and_publish_execution_result(%Wanda.Execution.Result{} = result) do
+    Wanda.Results.append(result)
```

Comment:
To keep consistency with the publishing error handling, maybe we should pattern match like:
`{:ok, _} = Wanda.Results.append(result)`

Reply:
not sure about this, as we bang in the context so it will raise an exception but it will never return an :ok tuple

Reply:
We are not banging in the context at the moment.

I can change the context to bang `Wanda.Repo.insert!()` and leave this line as it is.
Agreed?

Reply:
I just saw that the `append` method was returning the `{:ok, any}` or `{:error, any}` options, so to have consistency with the publish. No strong opinions though

Reply:
Banged with `Wanda.Repo.insert!()` in `Wanda.Results.save/1`